include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -30,6 +30,47 @@
 
 namespace seqan3
 {
+// -----------------------------------------------------------------------------
+// metafunction
+// -----------------------------------------------------------------------------
+
+//!\brief Transformation trait metafunction that returns the ungapped sequence type
+//!       (default: assume no gapped input and return the same type).
+//!\ingroup aligned_sequence
+template <typename t>
+struct ungapped_sequence_type
+{
+    //!\brief The transformation trait type.
+    using type = t;
+};
+
+//!\brief Helper typedef for the metafunction seqan3::ungapped_sequence_type
+template <typename t>
+using ungapped_sequence_type_t = typename ungapped_sequence_type<t>::type;
+
+/*!\brief Transformation trait metafunction that returns the ungapped sequence type.
+ * \ingroup aligned_sequence
+ * \tparam gapped_seq_type The gapped sequence container; must model seqan3::sequence_container_concept and its value
+ *                         type must be a seqan3::gapped alphabet.
```

Comment:
Hm, this is really not ideal. If it always returns a vector, it's not a generic metafunction and I would maybe just move it to detail and clearly state that it is a helper for verifying the concept. It's not used anywhere else, right?

Reply:
I agree. It is not used anywhere else yet.